lib/gears/filesystem.lua

```diff
@@ -11,20 +11,30 @@ local gtable = require("gears.table")
 
 local filesystem = {}
 
+local function make_parent(gfile)
+    local success, err = gfile:make_directory_with_parents()
+    if success then
+        return true
+    end
+    if err.domain == Gio.IOErrorEnum and err.code == "EXISTS" then
+        -- Directory already exists, let this count as success
+        return true
+    end
+    return false, err
+end
+
 --- Create a directory
 -- @tparam string dir The directory.
 -- @return (true, nil) on success, (false, err) on failure
 function filesystem.mkdir(dir)
```

Comment:
I'm not sure about code readability. `mkdir("foo/bar/baz", true)` isn't as self-explanatory as `make_parent_dir("foo/bar/baz")`.

Hm. Should `mkdir` be renamed to `make_directory` or something like that for consistency?